clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -82,32 +82,51 @@ class GCEUtil {
 
     def imageProjects = [projectName] + credentials?.imageProjects + baseImageProjects - null
     def sourceImage = null
+    // We want predictable iteration order that matches the order of insertion.
+    def imageProjectToNextPageTokenMap = new LinkedHashMap<String, String>()
 
-    def imageListBatch = buildBatchRequest(compute, clouddriverUserAgentApplicationName)
-    def imageListCallback = new JsonBatchCallback<ImageList>() {
-      @Override
-      void onFailure(GoogleJsonError e, HttpHeaders responseHeaders) throws IOException {
-        updateStatusAndThrowNotFoundException("Error locating $imageName in these projects: $imageProjects: $e.message", task, phase)
-      }
+    // This will ensure that each image project is queried.
+    imageProjects.each { imageProjectToNextPageTokenMap[it] = null }
 
-      @Override
-      void onSuccess(ImageList imageList, HttpHeaders responseHeaders) throws IOException {
-        // No need to look through these images if the requested image was already found.
-        if (!sourceImage) {
-          for (def image : imageList.items) {
-            if (image.name == imageName) {
-              sourceImage = image
-            }
+    while (!sourceImage && imageProjectToNextPageTokenMap) {
+      def imageListBatch = buildBatchRequest(compute, clouddriverUserAgentApplicationName)
+      def imageListCallback = new JsonBatchCallback<ImageList>() {
+        @Override
+        void onFailure(GoogleJsonError e, HttpHeaders responseHeaders) throws IOException {
+          updateStatusAndThrowNotFoundException("Error locating $imageName in these projects: $imageProjects: $e.message", task, phase)
```

Comment:
It feels a little odd to throw (rather than just log) the error here, since you might find the image in a different image project.